src/handlers/breeds.handler.ts

```diff
@@ -0,0 +1,64 @@
+import { APIGatewayEvent, Context, Handler } from 'aws-lambda'
+import { BreedsController } from '../controllers/breeds.controller'
+import { BreedsResponse } from '../domain/breed.domain'
+import { LogsService } from '../services/logs.service'
+import { CommonUtils } from '../utils/breeds.utils'
+
+const getBreedsHandler: Handler = async (
+  event: APIGatewayEvent,
+  context: Context,
+): Promise<BreedsResponse | Error> => {
+  let getBreedsResponse: BreedsResponse | Error
+  // warning log statement if lambda timeout is with in  3 seconds
+  const timer = CommonUtils.requestTimeWarning(context)
+
+  LogsService.log('info', `request event: ${JSON.stringify(event)}`)
+  try {
+    // controller call
+    getBreedsResponse = await new BreedsController().getBreeds()
+    // TODO: response sanitization for sensitive data
+    LogsService.log('info', `${context.functionName} resp: ${JSON.stringify(getBreedsResponse)}`)
+  } catch (error) {
+    LogsService.log('error', `api: ${context.functionName}, error:${error}`)
+    getBreedsResponse = {
+      statusCode: error.code || 500,
+      body: error.message || error,
+      message: 'failed',
+    }
+  } finally {
+    clearTimeout(timer)
+  }
+  return getBreedsResponse
+}
+
+// random breed handler
+async function getRandomBreedHandler(
```

Comment:
I can appreciate the consolidation of the service logic, but would also like to push further on the scalability for individual lambda handlers. Perhaps there is a better way to keep the existing directory structure you have, but have each handler in its own file? That would also separate the tests so each are independent of the other.

Reply:
sure, I can do that. I agree that it's better to have  lambda SoC ( separation of concerns).
To be honest I were having two lambda files initially but then opted out to be in single file.

Reply:
done. refactored lambda handlers.